hummingbot/cli/hummingbot_application.py

```diff
@@ -682,11 +672,18 @@ def start(self, log_level: Optional[str] = None):
                 if market is not None:
                     self.clock.add_iterator(market)
             self.clock.add_iterator(self.strategy)
-            
             self.strategy_task: asyncio.Task = asyncio.ensure_future(self.clock.run())
             self.app.log(f"\n  '{strategy_name}' strategy started.\n"
                          f"  You can use the `status` command to query the progress.")
+
             self.starting_balances = await self.wait_till_ready(self.balance_snapshot)
+            self.stop_loss_tracker = StopLossTracker(self.data_feed,
+                                                     list(self.assets),
```

Comment:
I'm not sure if we should include the base asset into the calculation of stop loss. In both XEMM and arbitrage... we're really just focusing on gaining more quote token while mostly ignoring price movements of the base token. If ETH drops by 10% while I'm market making on WETH-TUSD, and this thing stops my strategy due to the ETH movement... it may not be what the trader intended.

Reply:
I believe you and Mike have discussed the stop-loss design in this ticket (https://app.clubhouse.io/coinalpha/story/2947/add-kill-switch-parameter).

Basically, we let the user choose `fixed` or `dynamic` pricing. If fixed, we use the starting prices of tokens to evaluate gain/loss, this is ideal for XEMM and ARB and the base token doesn't really matter here. If dynamic, we would calculate gain/loss based by comparing starting prices and current prices -- this might be useful for Pure MM. In this case, the base token does matter.

Reply:
Ok, now I understand.